components/movementsensor/adxl345/adxl345.go

```diff
@@ -41,9 +41,54 @@ var modelName = resource.NewDefaultModel("accel-adxl345")
 
 // AttrConfig is a description of how to find an ADXL345 accelerometer on the robot.
 type AttrConfig struct {
-	BoardName              string `json:"board"`
-	I2cBus                 string `json:"i2c_bus"`
-	UseAlternateI2CAddress bool   `json:"use_alternate_i2c_address,omitempty"`
+	BoardName              string                 `json:"board"`
+	I2cBus                 string                 `json:"i2c_bus"`
+	UseAlternateI2CAddress bool                   `json:"use_alternate_i2c_address,omitempty"`
+	InterruptPin1          InterruptPinAttrConfig `json:"interrupt_pin1,omitempty"`
+	InterruptPin2          InterruptPinAttrConfig `json:"interrupt_pin2,omitempty"`
+}
+
+type InterruptPinAttrConfig struct {
+	EchoInterrupt      string              `json:"echo_interrupt_pin"`
+	SingleTap          bool                `json:"single_tap,omitempty"`
+	FreeFall           bool                `json:"free_fall,omitempty"`
+	TapAttrConfig      *TapAttrConfig      `json:"single_tap_attributes,omitempty"`
+	FreeFallAttrConfig *FreeFallAttrConfig `json:"free_fall_attributes,omitempty"`
+}
+
+type TapAttrConfig struct {
+	ExcludeTapX bool `json:"tap_x,omitempty"`
+	ExcludeTapY bool `json:"tap_y,omitempty"`
+	ExcludeTapZ bool `json:"tap_z,omitempty"`
+	ThreshTap   byte `json:"thresh_tap,omitempty"`
+	Dur         byte `json:"dur,omitempty"`
+}
+
+type FreeFallAttrConfig struct {
+	EchoInterrupt string `json:"echo_interrupt_pin"`
+	ThreshFF      byte   `json:"thresh_ff,omitempty"`
+	TimeFF        byte   `json:"time_ff,omitempty"`
+}
```

Comment:
How is this? I was getting stuck on determining the correct interrupt pin with the structure we discussed. In the way we discussed we would have the user put the pin on the `TapAttrConfig` or `FreeFallAttrConfig` but I would have no way of mapping whatever they named the interrupt pin to INT1 or INT2 on the chip.

This maintains the nesting of the configs associated with interrupts but does it in the pin config instead of the top level, so that the user has to specify what they are naming INT1 and INT2. Thoughts?